### PR TITLE
Migrate CodeCache switching logic to CodeGenerator

### PR DESCRIPTION
**Pull request by 0xdaryl**
* Move NumReservedIPICTrampolines getter/setters to X86 CodeGenerator
* Move codeCacheSwitched getter/setter to CodeGenerator
* Implement switchCodeCacheTo() in CodeGenerator

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Note to reviewers:

This PR is written in such a way as to prevent breakages in downstream projects.  Once this PR is merged and the necessary changes made to known downstream projects then the (hopefully obvious) scaffolding will be removed in OMR.

One of the by-products of the scaffolding left in place for now are the `TR_TARGET_X86` macros that guard calls to CodeGenerator `setNumReservedIPICTrampolines` and `getNumReservedIPICTrampolines` in some places.  The correct way to fix this would be to provide X86-specialized classes but in the case of the Compilation class there isn't already an X86 specialization.  This means that by adding one would cause a downstream breakage anyway because its build system would need to be modified to be aware of the new file.  Since that isn't in the spirit of the scaffolding, I simply guarded the calls with the target macro to avoid introducing a new file.  This code should be extremely transient.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build all

File: compiler/compile/OMRCompilation.cpp [hunk 1]
```diff
@@ -2739,3 +2742,29 @@ bool OMR::Compilation::incompleteOptimizerSupportForReadWriteBarriers()
...
+   return self()->cg()->getNumReservedIPICTrampolines();
+#else
+   return 0;
+#endif
+   }
+
+void OMR::Compilation::setNumReservedIPICTrampolines(int32_t n)
+   {
+#ifdef TR_TARGET_X86
+   return self()->cg()->setNumReservedIPICTrampolines(n);
```
Comment (Leonardo2718) [inline compiler/compile/OMRCompilation.cpp]: I guess this technically works, but given that the function returns void, would it make more sense to skip the `return`?

```suggestion
   self()->cg()->setNumReservedIPICTrampolines(n);
```

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**